src/main/kotlin/org/jqassistant/tooling/intellij/plugin/editor/templates/RuleJqaTemplateFileCreation.kt

```diff
@@ -92,26 +73,63 @@ abstract class RuleJqaTemplateFileCreator(
         } ?: Messages.showErrorDialog(project, "File created but could not be opened.", "Error")
     }
 
-    // Prompts the user for input using a dialog box and returns the input.
-    private fun getUserInput(project: Project, message: String): String? =
-        Messages.showInputDialog(project, message, "Input", Messages.getQuestionIcon())?.trim()
+    // Prompts the user for input and validates the file name recursively.
+    private fun getValidFileName(project: Project, message: String?, initialValue: String?): String? {
+        var input = getUserInput(project, message ?: "Enter file name:", initialValue) ?: return null
+
+        if (input.isEmpty()) {
+            Messages.showWarningDialog(project, "File name can't be empty.", "Invalid Input")
+            return null
+        } else if (!isNameValid(input)) {
+            Messages.showWarningDialog(project, "Invalid file name.", "Invalid Input")
```

Comment:
I think it users should be informed what exactly is wrong with their file name,  or at least the message could list which characters are forbidden. You could also check out [`PathUtilRt.suggestFileName`](https://github.com/JetBrains/intellij-community/blob/273bd2a4cb13babb6b7f10b990dcca432f367233/platform/util-rt/src/com/intellij/util/PathUtilRt.java#L120) to give the user direct feedback